lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php

```diff
@@ -1803,6 +1803,10 @@ public function getSingleIdentifierFieldName()
         if ($this->isIdentifierComposite) {
             throw MappingException::singleIdNotAllowedOnCompositePrimaryKey($this->name);
         }
+        
+        if ( ! isset($this->identifier[0])) {
```

Comment:
Can you also mention this change in the docblock? We currently have `@throws MappingException If the class has a composite primary key.` but would be good to say something like `@throws MappingException If the class doesn't have an identifier or it has a composite primary key.`